packages/query/src/index.ts

```diff
@@ -415,27 +378,30 @@ export class LiveQuery implements WithTx, Client {
         console.log('failed to update Live Query: ', err)
       })
 
-    this.queries.set(_class, queries)
-    while (this.queue.length > CACHE_SIZE) {
+    if (this.queue.size > CACHE_SIZE) {
       this.remove()
     }
     return q
   }
 
   private remove (): void {
-    const q = this.queue.shift()
-    if (q === undefined) return
-    const queries = this.queries.get(q._class)
-    const pos = queries?.indexOf(q) ?? -1
-    if (pos >= 0 && queries !== undefined) {
-      queries.splice(pos, 1)
+    const used = Array.from(this.queue.values()).sort((a, b) => a.lastUsed - b.lastUsed)
+    for (let i = 0; i < CACHE_SIZE / 10; i++) {
```

Comment:
Why /10?

Reply:
To not remove every time, but from time to time